source/mainnet/smart-contracts/tutorials/wCCD/wCCD-introduction.rst

```diff
@@ -0,0 +1,97 @@
+    - `multi <https://github.com/Concordium/concordium-rust-smart-contracts/blob/main/examples/cis2-multi/src/lib.rs>`_
+
+
+Implementing two interfaces (one for CCD and another for ``CIS-2`` tokens)
+within the same smart contract can be cumbersome for developers and adds
+complexity. We need a process that converts CCD into a token (named wCCD) that is ``CIS-2``
+compliant so dApps can interact with it easily. For example, decentralized
+exchanges depend on the wCCD token because the wCCD token can be traded with other ``CIS-2`` tokens.
+
+Wrapping CCD refers to the process of converting the native currency CCD into
+a ``CIS-2`` compliant token (wCCD) at a 1:1 ratio by sending CCD to the wCCD smart
+contract and getting wCCD in return. The ``wrap`` function accepts an amount of CCD and mints(creates)
+the same amount of wCCD tokens. It takes a receiving address as the parameter and transfers
+the minted amount of wCCD tokens to this receiver.
+
+Unwrapping CCD refers to the opposite process of converting the ``CIS-2``
+compliant wCCD token at a 1:1 ratio back to the native currency CCD by burning(destroying) the
+wCCD token in the wCCD smart contract and getting CCD in return.
+The ``unwrap`` function takes the amount of tokens to unwrap and burns(destroys) them
+before transferring the same amount of CCD to the receiver.
+
+The circulating supply of the wCCD token is backed 1:1
+by the CCD balance on the wCCD smart contract.
+
+Protocol contract addresses
+---------------------------
+
+The Concordium foundation maintains the canonical wCCD smart contract protocol and promotes its
+usage to create a coherent overall smart contract ecosystem on the Concordium blockchain.
+
+.. note::
+
+    The below testnet addresses are an early version of the wCCD contract protocol.
+    They will be updated when the protocol is finalized and some minor changes can be expected.
+
+The canonical wCCD smart contract protocol following the ``CIS-2`` standard
+is deployed on ``testnet`` at the following addresses:
+
++-----------------------------------------------------+-------------------------------------------------+
+| ``ProxyContractAddress (Testnet):``                 |  { index: 652, subindex: 0 }                    |
++-----------------------------------------------------+-------------------------------------------------+
+| ``ImplementationContractAddress (Testnet):``        |  { index: 651, subindex: 0 }                    |
++-----------------------------------------------------+-------------------------------------------------+
+| ``StateContractAddress (Testnet):``                 |  { index: 650, subindex: 0 }                    |
++-----------------------------------------------------+-------------------------------------------------+
+|                                                     |                                                 |
```

Comment:
Why is there an empty bottom row in these tables? Is it to add a bottom line in the table?

Reply:
yes, otherwise there was no bottom line in the table.